api_tests/base/test_auth.py

```diff
@@ -195,3 +196,23 @@ def test_node_write_scope_cant_read_user_view(self, mock_user_info):
 
         res = self.app.get(url, params=payload, auth='some_valid_token', auth_type='jwt', expect_errors=True)
         assert_equal(res.status_code, 403)
+
+    @mock.patch('framework.auth.cas.CasClient.profile')
+    def test_full_read_scope_can_read_guid_view(self, mock_user_info):
+        project = ProjectFactory()
+        mock_user_info.return_value = self._scoped_response(['osf.full_read'])
+        url = api_v2_url('guids/{}/'.format(project._id), base_route='/', base_prefix='v2/')
+        res = self.app.get(url, auth='some_valid_token', auth_type='jwt')
+        redirect_url = '{}{}nodes/{}/'.format(API_DOMAIN, API_BASE, project._id)
+        assert_equal(res.status_code, 302)
+        assert_equal(res.location, redirect_url)
```

Comment:
What I meant was that we should have the user redirect to a node that they have permissions for and ensure, using a single request, that the node information loads. So:

- User creates node with guid X
- User goes to /v2/guids/X using their auth token with follow_redirect=true
- User gets the proper project title with no more work

However, since you mention it, you should probably also do that where they are given a 403/401 when appropriate. If that works, I'm not super concerned about scopes. But if you want to go mad with power, see what happens if their token has scope for the redirect but not for the resource behind it.